utilities/postgres.go

```diff
@@ -0,0 +1,67 @@
+package utilities
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgerrcode"
+)
+
+// PostgresError is a custom error struct for marshalling Postgres errors to JSON.
+type PostgresError struct {
+	Code           string `json:"code"`
+	HttpStatusCode int    `json:"-"`
+	Message        string `json:"message"`
+	Hint           string `json:"hint,omitempty"`
+	Detail         string `json:"detail,omitempty"`
+}
+
+// NewPostgresError returns a new PostgresError if the error was from a publicly
+// accessible Postgres error.
+func NewPostgresError(err error) *PostgresError {
+	var pgErr *pgconn.PgError
+	if errors.As(err, &pgErr) && isPubliclyAccessiblePostgresError(pgErr.Code) {
+		return &PostgresError{
+			Code:           pgErr.Code,
+			HttpStatusCode: getHttpStatusCodeFromPostgresErrorCode(pgErr.Code),
+			Message:        pgErr.Message,
+			Detail:         pgErr.Detail,
+			Hint:           pgErr.Hint,
+		}
+	}
+
+	return nil
+}
+
+// isPubliclyAccessiblePostgresError checks if the Postgres error should be
+// made accessible.
+func isPubliclyAccessiblePostgresError(code string) bool {
+	if len(code) != 5 {
+		return false
+	}
+
+	if code == pgerrcode.RaiseException {
+		return true
+	}
+
+	return getHttpStatusCodeFromPostgresErrorCode(code) != 0
```

Comment:
is there a reason we check against "0" or is "0" just an arbitrary number picked that isn't mapped to any error codes?

Reply:
Yeah, `0` is the default `int` value returned by `getHttpStatusCodeFromPostgresErrorCode` if it is unable to determine an appropriate HTTP status code from the given Postgres error code.

Reply:
nice, could we add a comment just to make it explicit that this is just the default int value we decided to return?